patches/tModLoader/Terraria.ModLoader/ModPlayer.cs

```diff
@@ -815,5 +815,57 @@ public class ModPlayer
 		public virtual bool ShiftClickSlot(Item[] inventory, int context, int slot) {
 			return false;
 		}
+
+		/// <summary>
+		/// Called whenever the player sells an item to an NPC.
+		/// </summary>
+		/// <param name="vendor"></param>
+		/// <param name="item"></param>
+		public virtual void PostSellItem(NPC customer, Item item)
+		{
+
+		}
+
+		/// <summary>
+		/// Return false to prevent a transaction. Called before the transaction.
+		/// </summary>
+		/// <param name="customer"></param>
+		/// <param name="item"></param>
+		/// <returns></returns>
+		public virtual bool ModifySellItem(NPC customer, ref Item item) => true;
+
+		/// <summary>
+		/// Called whenever the player buys an item from an NPC.
+		/// </summary>
+		/// <param name="vendor"></param>
+		/// <param name="item"></param>
+		public virtual void PostBuyItem(NPC vendor, Item item) {
+
+		}
+
+		/// <summary>
+		/// Return false to prevent a transaction. Called before the transaction.
+		/// </summary>
+		/// <param name="vendor"></param>
+		/// <param name="item"></param>
+		/// <returns></returns>
+		public virtual bool ModifyBuyItem(NPC vendor, Item item) => true;
```

Comment:
The item param should be ref.

Reply:
I wonder if I should add a manual “ref int price” param as well.

Reply:
No, the Item parameter does not need the `ref` keyword, it is already a reference. For an int param, yes it would need `ref`, since the primitive type 'int' is copied, not referenced. But no, you should not add it (as mentioned by chicken-bones), this hook should not be for modifying prices directly.